libgambatte/include/gambatte-c.h

```diff
@@ -16,145 +16,171 @@
 //   51 Franklin St, Fifth Floor, Boston, MA  02110-1301, USA.
 //
 
-/* C interface for using a shared libgambatte */
+/* C interface for using libgambatte */
 
 #ifndef GAMBATTE_C_H
 #define GAMBATTE_C_H
 
 #include "config.h"
 
+#ifndef __cplusplus
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct GB GB;
+typedef struct FPtrs FPtrs;
```

Comment:
FPtrs is expected to be constructed from the user side rather than the dll side, so it should be fully defined here.